client/lib/js/utils/cryptoconditions/filter.js

```diff
@@ -0,0 +1,24 @@
+
+const filterByType = ({ condition, typeId, maxDepth }) => {
+    let res = [];
+    if (condition.hasOwnProperty('type_id') && condition.type_id === typeId) {
+        res.push(condition);
+    }
+
+    const nextMaxDepth = (maxDepth === parseInt(maxDepth, 10)) ? maxDepth - 1 : null;
```

Comment:
so, yeah... I don't know either what I am doing there :-D

ah, maybe it has an `undefined` when not provided as an arg... and wanted to catch that as being the same as `null`